lib/_tls_wrap.js

```diff
@@ -925,7 +926,9 @@ function tlsConnectionListener(rawSocket) {
     handshakeTimeout: this[kHandshakeTimeout],
     ALPNProtocols: this.ALPNProtocols,
     SNICallback: this[kSNICallback] || SNICallback,
-    enableTrace: this[kEnableTrace]
+    enableTrace: this[kEnableTrace],
+    allowHalfOpen: this.allowHalfOpen,
```

Comment:
I believe that previously `tlsOptions.allowHalfOpen` was always undefined.

Reply:
Can you add a test for this then?

Reply:
Also, maybe either change the commit message or split this change out since it's unrelated to `pauseOnConnect`?

Reply:
Should already be handled here https://github.com/nodejs/node/blob/9f830f37daf55a2c903153e1681873a8cd61baee/lib/_tls_wrap.js#L413